src/Refinery/To/Transformation/DateTimeTransformation.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/* Copyright (c) 2018 Nils Haagen <nils.haagen@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\Refinery\To\Transformation;
+
+use ILIAS\Data\Factory;
+use ILIAS\Data\Result;
+use ILIAS\Refinery\Transformation;
+
+/**
+ * Transform value to php \DateTimeImmutable
+ */
+class DateTimeTransformation implements Transformation {
+
+	/**
+	 * @var DataFactory
+	 */
+	private $factory;
+
+	/**
+	 * @param Factory $factory
+	 */
+	public function __construct(Factory $factory)
+	{
+		$this->factory = $factory;
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function transform($from) {
+		if($from) {
+			$result = $this->attemptTransformation($from);
+			if($result->isError()) {
+				throw new \InvalidArgumentException($result->error(), 1);
+			}
+			return $result->value();
+		}
+		return null;
```

Comment:
This is not good. The `Refinery` should allow people to expect, that this always returns a `DateTimeImmutable`, which isn't the case here. This seems to be a requirement from the input and thus should be moved to that layer.